src/CommandLine/Core/TokenPartitioner.cs

```diff
@@ -23,11 +24,11 @@ public static
             var scalars = Scalar.Partition(tokenList, typeLookup).Memorize();
             var sequences = Sequence.Partition(tokenList, typeLookup).Memorize();
             var nonOptions = tokenList
-                .Where(t => !switches.Contains(t))
-                .Where(t => !scalars.Contains(t))
-                .Where(t => !sequences.Contains(t)).Memorize();
+                .Where(t => !switches.Contains(t, ReferenceEqualityComparer.Default))
+                .Where(t => !scalars.Contains(t, ReferenceEqualityComparer.Default))
+                .Where(t => !sequences.Contains(t, ReferenceEqualityComparer.Default)).Memorize();
             var values = nonOptions.Where(v => v.IsValue()).Memorize();
-            var errors = nonOptions.Except(values).Memorize();
+            var errors = nonOptions.Except(values, ReferenceEqualityComparer.Default).Cast<Token>().Memorize();

```

Comment:
I realized that I can make other cast - `(IEqualityComparer<Token>)ReferenceEqualityComparer.Default` so `Cast<Token>` will not be needed. Or can make modify comparer to classic generic version - `ReferenceEqualityComparer<T>`. Don't want to inherit `IEqualityComparer<Token>`, it's too concrete for this task.

Will replace `==` with `ReferenceEquals`.